python/dask_cudf/dask_cudf/backends.py

```diff
@@ -493,11 +494,15 @@ def read_parquet(*args, engine=None, **kwargs):
             )
 
         @staticmethod
-        def read_json(*args, engine=None, **kwargs):
+        def read_json(*args, engine="cudf", **kwargs):
             return _default_backend(
                 dd.read_json,
                 *args,
-                engine=cudf.read_json,
+                engine=(
+                    partial(cudf.read_json, engine=engine)
+                    if isinstance(engine, str)
+                    else engine
+                ),
```

Comment:
So a simpler patch would just be passing `engine=engine or cudf.read_json` through. Though I suppose that this is a slightly easier to use.

Would be useful to add a short docstring describing the `engine` parameter (since it now differs from `dask.dataframe.read_json`.

Question: what happens if the user passes `pd.read_json` as engine here?

Reply:
Thanks for the quick review @wence- !

>So a simpler patch would just be passing engine=engine or cudf.read_json through. Though I suppose that this is a slightly easier to use.

Not sure this is true?

Just to clarify what is intended here: Before this change, `dd`/`dask_cudf` was expecting a callable argument for `engine` (like `cudf.read_json` or `pd.read_json`).  The `cudf` version, however, expects a string argument (like "cudf", "cudf_experimental", or "pandas"). The purpose of this change is to let the user pass in either to `dask_cudf` (or to the "cudf"-dispatched version of `dd.read_json`)

>Would be useful to add a short docstring describing the engine parameter (since it now differs from dask.dataframe.read_json.

I was hoping to avoid this originally, but I completely agree.

>Question: what happens if the user passes pd.read_json as engine here?

The user will get a pandas-backed Dask-DataFrame. The "confusing" result is if the user passes in "pandas", because `cudf.read_json` will take this to mean: "use pandas to read the data, but then convert to cudf".

Reply:
> Thanks for the quick review @wence- !
> 
> > So a simpler patch would just be passing engine=engine or cudf.read_json through. Though I suppose that this is a slightly easier to use.
> 
> Not sure this is true?

Ah sorry, I meant:

```python
def read_json(..., engine=None, ...):
    return whatever.read_json(..., engine=engine or cudf.read_json, ...)
```
And if the user wants to provide a default to `cudf.read_json` then they would say:
```python
read_json(..., engine=partial(cudf.read_json, engine="cudf_experimental"), ...)
```

But this is not very nice UX, so this is basically a long way of saying I approve of this change.

 [...]

> > Would be useful to add a short docstring describing the engine parameter (since it now differs from dask.dataframe.read_json.
> 
> I was hoping to avoid this originally, but I completely agree.
> 
> > Question: what happens if the user passes pd.read_json as engine here?
> 
> The user will get a pandas-backed Dask-DataFrame. The "confusing" result is if the user passes in "pandas", because `cudf.read_json` will take this to mean: "use pandas to read the data, but then convert to cudf".

Thanks, I think the docstring clarifies nicely what is going to happen.